scipy/optimize/tests/test_minpack.py

```diff
@@ -453,6 +453,25 @@ def f(x, a, b):
         assert_raises(ValueError, curve_fit, f, xdata, ydata, bounds=bounds,
                       method='lm')
 
+    def test_bounds_p0(self):
+        # This test is for issue #5719. The problem was that an initial guess
+        # was ignored when 'trf' or 'dogbox' methods were invoked.
+        def f(x, a, b):
+            return a * np.exp(-b*x)
+
+        xdata = np.linspace(0, 1, 11)
+        ydata = f(xdata, 2., 2.)
+        bounds = ([1., 0], [1.5, 3.])
+        p0 = [1.01, 2.9]
+        for method in [None, 'trf', 'dogbox']:
+            popt_1, _ = curve_fit(f, xdata, ydata, bounds=bounds,
+                                  method=method)
+            popt_2, _ = curve_fit(f, xdata, ydata, p0, bounds=bounds,
+                                  method=method)
+            # Test that popt_1 and popt_2 are not too close. A bit weird
+            # to test like this, but probably it's the only way.
+            assert_(not np.allclose(popt_1, popt_2, rtol=1e-8))
```

Comment:
Agree, this looks a little weird :-). How about making it a little more robust:

```
In [30]: x = np.linspace(-2*np.pi, 2*np.pi, 40)

In [31]: y = np.sin(x)

In [32]: def f(x, a): return np.sin(x + a)
```

On master:

```
In [33]: curve_fit(f, x, y, p0=1.1*np.pi)
Out[33]: (array([ 6.28318531]), array([[  1.74886865e-33]]))

In [34]: curve_fit(f, x, y, p0=1.1*np.pi, method='trf')
Out[34]: (array([  1.43099780e-11]), array([[  5.25070566e-24]]))
```

On branch: 

```
In [13]: curve_fit(f, x, y, p0=1.1*np.pi)
Out[13]: (array([ 6.28318531]), array([[  1.74886865e-33]]))

In [14]: curve_fit(f, x, y, p0=1.1*np.pi, method='trf')
Out[14]: (array([ 6.28318531]), array([[  1.74886865e-33]]))
```